### PR TITLE
[Merged by Bors] - feat: port Data.ZMod.Quotient

### PR DESCRIPTION
**Pull request by eric-wieser**
---

- [x] depends on: #4050

[Open in Gitpod](https://gitpod.io/from-referrer/)

This replaces #3333

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#4050
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/ZMod/Quotient.lean [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+/-!
+# `ZMod n` and quotient groups / rings
+
+This file relates `ZMod n` to the quotient group
+`quotient_add_group.quotient (AddSubgroup.zmultiples n)` and to the quotient ring
+`(Ideal.span {n}).quotient`.
+
+## Main definitions
+
+ - `zmod.quotient_zmultiples_nat_equiv_zmod` and `zmod.quotient_zmultiples_equiv_zmod`:
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/ZMod/Quotient.lean:26]: Comments will need to be fixed before merging

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#4050~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/ZMod/Quotient.lean [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+open QuotientAddGroup
+
+open ZMod
+
+variable (n : ℕ) {A R : Type _} [AddGroup A] [Ring R]
+
+namespace Int
+
+/-- `ℤ` modulo multiples of `n : ℕ` is `ZMod n`. -/
+def quotientZmultiplesNatEquivZmod : ℤ ⧸ AddSubgroup.zmultiples (n : ℤ) ≃+ ZMod n :=
```
Comment (Vierkantor) [inline Mathlib/Data/ZMod/Quotient.lean:49]: Shouldn't this be `ZMod`? (Everywhere in this file.) I would also suggest `ZMultiples` but it seems `Zmultiples` is used consistently in other files.
```suggestion
def quotientZmultiplesNatEquivZMod : ℤ ⧸ AddSubgroup.zmultiples (n : ℤ) ≃+ ZMod n :=
```

**Review by Vierkantor — APPROVED** [review]
LGTM otherwise, thanks!

bors d+

**Pull request closed without merging**